src/BDCS/API/V0.hs

```diff
@@ -129,3 +135,152 @@ errTest = throwError myError
+-- >          "jboss",
+-- >          "kubernetes",
+-- >          "octave",
+-- >      ],
+-- >      "offset": 0,
+-- >      "limit": 20,
+-- >      "total": 6
+-- >  }
+recipesList :: GitLock -> T.Text -> Handler RecipesListResponse
+recipesList repoLock branch = liftIO $ RWL.withRead (gitRepoLock repoLock) $ do
+    -- TODO Figure out how to catch GitError and throw a ServantErr
+    recipes <- liftIO $ listBranchFiles (gitRepo repoLock) branch
+    return $ RecipesListResponse recipes 0 0 (length recipes)
+  where
+    handleGitErrors :: GitError -> ServantErr
+    handleGitErrors e = createApiError err500 "recipes_list" ("Git Error: " ++ show e)
+
+
+data WorkspaceChanges = WorkspaceChanges {
+    name        :: T.Text,
+    changed     :: Bool
+} deriving (Generic, Show, Eq)
+instance ToJSON WorkspaceChanges
+instance FromJSON WorkspaceChanges
+
+data RecipesInfoResponse = RecipesInfoResponse {
+    changes     :: [WorkspaceChanges],
+    recipes     :: [Recipe]
+} deriving (Generic, Show, Eq)
+instance ToJSON RecipesInfoResponse
+instance FromJSON RecipesInfoResponse
+
+-- | /api/v0/recipes/info/<recipes>
+-- Return the contents of the recipe, or a list of recipes
+--
+-- > {
+-- >     "changes": [
+-- >         {
+-- >             "name": "recipe-test",
+-- >             "changed": true
+-- >         },
+-- >     ],
+-- >     "recipes": [
+-- >         {
+-- >             "name": "http-server",
+-- >             "description": "An example http server with PHP and MySQL support.",
+-- >             "version": "0.0.1",
+-- >             "modules": [
+-- >                 {
+-- >                     "name": "httpd",
+-- >                     "version": "2.4.*"
+-- >                 },
+-- >                 {
+-- >                     "name": "mod_auth_kerb",
+-- >                     "version": "5.4"
+-- >                 },
+-- >                 {
+-- >                     "name": "mod_ssl",
+-- >                     "version": "2.4.*"
+-- >                 },
+-- >                 {
+-- >                     "name": "php",
+-- >                     "version": "5.4.*"
+-- >                 },
+-- >                 {
+-- >                     "name": "php-mysql",
+-- >                     "version": "5.4.*"
+-- >                 }
+-- >             ],
+-- >             "packages": [
+-- >                 {
+-- >                     "name": "tmux",
+-- >                     "version": "2.2"
+-- >                 },
+-- >                 {
+-- >                     "name": "openssh-server",
+-- >                     "version": "6.6.*"
+-- >                 },
+-- >                 {
+-- >                     "name": "rsync",
+-- >                     "version": "3.0.*"
+-- >                 }
+-- >             ]
+-- >         }
+-- >     ],
+-- > }
+-- 
+recipesInfo :: GitLock -> T.Text -> T.Text -> Handler RecipesInfoResponse
+recipesInfo repoLock branch recipes = liftIO $ RWL.withRead (gitRepoLock repoLock) $ do
+    undefined
```

Comment:
Something for the future, or something that was meant to go into this patch?

Reply:
Future. Next step is getting the /recipes/* routes working.